### PR TITLE
add a-tiny-js-world task

### PR DESCRIPTION
**Pull request by frostwolm**
_Complete the form below and remove this line._

### A Tiny JS World

[Demo](https://frostwolm.github.io/a-tiny-JS-world/) |
[Code base](https://github.com/frostwolm/a-tiny-JS-world)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/frostwolm/a-tiny-js-world/index.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+   print('<strong>ABC</strong>');
+   print('<strong>ABC</strong>', 'div');
+
+   print('human; John; male; 2; 2; Hello world!; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny', 'div');
+   */
+function getStrFromObj(obj) {
+  let arr = [];
+  Object.values(obj).forEach((item) => {
```
Comment (OleksiyRudenko) [inline submissions/frostwolm/a-tiny-js-world/index.js]: This code fragment takes each element from an array, transforms the element, and adds the transformed element to a resulting array.
A clear pattern for `#Array.map`, not `#Array.forEach`.
Please, refactor using `.map()`.

File: submissions/frostwolm/a-tiny-js-world/index.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    if (Array.isArray(item)) {
+      let names = [];
+      item.forEach((frnd) => {
```
Comment (OleksiyRudenko) [inline submissions/frostwolm/a-tiny-js-world/index.js]: When arrow function `=>` has a single parameter you may skip surrounding parentheses. This makes code better readable a bit.

File: submissions/frostwolm/a-tiny-js-world/index.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        names.push(frnd.name);
+      });
+      arr.push(names.join(', '));
+    }else{
+      arr.push(item);
+    }
+  });
+  return arr.join('; ');
```
Comment (OleksiyRudenko) [inline submissions/frostwolm/a-tiny-js-world/index.js]: If you employ `.map()` above you will also benefit from chaining `.join()`

File: submissions/frostwolm/a-tiny-js-world/index.js (hunk 1 above)
Comment (OleksiyRudenko) [reply submissions/frostwolm/a-tiny-js-world/index.js]: Here you also rely on `obj` properties order, which is not guaranteed.
Should someone define next object where `hands` property is defined before `legs` property the resulting output won't be as expected.
Please, ensure that properties are processed in an explicitly specified order.

**Comment by OleksiyRudenko** [comment]
Please resolve conversations that has been processed. This gives a reviewer an idea of your progress.

**Comment by OleksiyRudenko** [comment]
@frostwolm Sorry for I missed to check if you have reacted to the review.
What I see is that your PR contains code from other contributors. Some files are removed.
We cannot merge this PR as it will affect other contributor's code and will remove some contributions that are out of scope.
You do not need to merge master into your feature branch.

Can you please create a new feature branch when on master and also from a commit that comes before any of your commits, and publish your `index.js` there?

If you need any support on this, please, let me know.

**Comment by zonzujiro** [comment]
Looks like that PR was abandoned by the author. Therefore I will close it but we always can open it again.

**Pull request closed without merging**